autonat.go

```diff
@@ -277,65 +327,98 @@ func (as *AmbientAutoNAT) scheduleProbe() time.Duration {
 
 // Update the current status based on an observed result.
 func (as *AmbientAutoNAT) recordObservation(observation autoNATResult) {
-	currentStatus := as.status.Load().(autoNATResult)
-	if observation.Reachability == network.ReachabilityPublic {
-		log.Debugf("NAT status is public")
-		changed := false
-		if currentStatus.Reachability != network.ReachabilityPublic {
-			// we are flipping our NATStatus, so confidence drops to 0
-			as.confidence = 0
-			if as.service != nil {
-				as.service.Enable()
-			}
-			changed = true
-		} else if as.confidence < 3 {
-			as.confidence++
+	// record observation
+	for addr, reach := range observation.addrReachability {
+		switch reach {
+		case network.ReachabilityPublic:
+			as.publicAddrsManager.record(observation.observer, addr)
```

Comment:
presumably you also want to record inbound dials normally received and not just as a result of autonat? how does that work?